scripts/apps/archive/index.js

```diff
@@ -128,8 +128,15 @@ angular.module('superdesk.apps.archive', [
                 label: gettext('Duplicate'),
                 icon: 'copy',
                 monitor: true,
-                controller: ['api', 'notify', '$rootScope', 'data', function(api, notify, $rootScope, data) {
-                    api.save('duplicate', {}, {desk: data.item.task.desk}, data.item)
+                controller: ['api', 'notify', '$rootScope', 'data', 'desks', '$location', 'workspaces', 'session',
```

Comment:
I know you're just extending what has already been here, but since this function has now grown, would you mind adding it into a separate file? For example `apps/archive/controllers/DuplicateController.js` might work.